### PR TITLE
Remove hardcoded default for url_base

### PR DESCRIPTION
**Pull request by adriansuarez**
This change removes the hardcoded default for url_base of http://localhost:8080 and performs validation on the url_base value.

### CODE REVIEW

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.

File: Makefile [hunk 1]
```diff
@@ -197,7 +197,7 @@ coverage-report:
 .PHONY: integration-tests
 integration-tests: $(TERRAFORM_BIN) $(KUBECTL_BIN) ## Start test environment, run acceptance tests, generate coverage report, and teardown test environment
 	$(MAKE) deploy-test-helper
-	KUBECONFIG=$(OUTPUT_DIR)/kubeconfig.yml $(MAKE) testacc
```
Comment (kontaras) [inline Makefile:200]: Does it make sense to only set `NUODB_CP_URL_BASE` if it is not already set? We are already letting through `NUODB_CP_USER` and `NUODB_CP_PASSWORD`, if the user sets them (which they should not), so it is weird to only override one of the settings.

File: internal/provider/provider.go [hunk 2]
```diff
@@ -85,27 +88,27 @@ func (p *NuoDbaasProvider) Metadata(ctx context.Context, req provider.MetadataRe
 
 func (p *NuoDbaasProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		Description: "The NuoDB DBaaS provider provides the ability to manage the projects and databases running under the NuoDB Control Plane.",
+		Description: "The NuoDB DBaaS Provider manages NuoDB databases using the NuoDB Control Plane.",
```
Comment (kontaras) [inline internal/provider/provider.go:91]: Why drop "projects" from the description? They are an implementation detail for most users, but it is still nice to have it noted.

**Review by kontaras — APPROVED** [review]

File: internal/provider/provider.go (hunk 2 above)
Comment (adriansuarez) [reply internal/provider/provider.go:91]: This is just a one sentence description of what the provider does, and I wanted to be concise. We will add resources for backups and backup policies as well eventually. We don't need to list every resource in this short description.

File: internal/provider_test/integration_test.go [hunk 3]
```diff
@@ -854,6 +853,30 @@ func TestNegative(t *testing.T) {
 		require.Contains(t, string(out), `missing unit in duration "999"`)
 		require.Contains(t, string(out), "Timeout for database update is negative: -1s")
 		require.Contains(t, string(out), "Invalid resource type: badresource")
+		vars.providerCfg.Timeouts = nil
+
+		// Temporarily override environment variable NUODB_CP_URL_BASE
+		// so that URL is not specified at all
+		urlBase := os.Getenv(NUODB_CP_URL_BASE)
+		defer os.Setenv(NUODB_CP_URL_BASE, urlBase)
+		os.Unsetenv(NUODB_CP_URL_BASE)
```
[the remark this answers is not in the document]
Comment (adriansuarez) [reply internal/provider_test/integration_test.go:862]: You mean setting `NUODB_CP_URL_BASE=""`?

The default is resolved using `os.Getenv()` in the provider configuration, which returns an empty string for unspecified environment variables. So I don't think we can more test coverage by setting it to empty explicitly.
